### PR TITLE
Added dockerfile and docker-compose.yaml

### PR DESCRIPTION
**Pull request by Saif-Uz-Zaman**

### CODE REVIEW

**Comment by Saif-Uz-Zaman** [comment]
@dogukanteber I am going to modify the README.md

1. For `linux` machine you can set the `network_name=host` in docker.env
2. For docker desktop in `Mac` or `Windows` you can set the custom network network_name=custom_network_name in docker.env.
(N.B. If you facing issue like ` network custom_network_name not found` You have to create it like `docker network create  custom_network_name` )

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
+!/bin/bash
```
Comment (dogukanteber) [inline docker-entrypoint.sh:1]: `#!/bin/bash`

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+Reset existing database
```
Comment (dogukanteber) [inline docker-entrypoint.sh:3]: `#Reset existing database`

**Comment by dogukanteber** [comment]
Thanks for the update @Saif-Uz-Zaman. I got rid of that error. Now, the database and the app is running but this time I got `FATAL:  password authentication failed for user "postgres"` error.

This is my `docker.env` file:

```
NETWORK_NAME=platform_test
DBUSER=postgres
DBPASS=postgres
DBNAME=ou_db
```

I got the same error with these variables:

```
NETWORK_NAME=platform_test
DBUSER=admin
DBPASS=password
DBNAME=superagent
```

**Comment by Saif-Uz-Zaman** [comment]
@dogukanteber possibly it is all because of docker volume caching. Just try this command, Hopefully it will work
```
docker compose --env-file docker.env down
docker volume prune
docker compose --env-file docker.env up --build
```

**Comment by dogukanteber** [comment]
Unfortunately, that was not the case. I also searched online but could not get rid of this issue. I asked @adrianmcphee to run the container on his machine, to make sure this error is not related to me. In the meantime, could you double-check the compose file?

I appreciate your efforts

**Comment by adrianmcphee** [comment]
Hi @Saif-Uz-Zaman  and @dogukanteber 

Firstly, thank you and great work Saif! 

Once I deleted the NETWORK_MODE setting from docker.env then it ran, as below.

What I find a bit strange is the end result... despite the claim "Starting development server at http://0.0.0.0:80/", if I go to that address then I have nothing.  I'm not planning to use docker for local development as we it is deliberately simple as is, and do don't have fancy dependencies (any more) etc. - however for our target build/deploy processes this is very nice.

[+] Running 3/3
 ⠿ Network platform_default  Created                                                                                                                                                                                    0.0s
 ⠿ Container pgdb            Recreated                                                                                                                                                                                  0.1s
 ⠿ Container platform        Recreated                                                                                                                                                                                  0.1s
Attaching to pgdb, platform
pgdb      | The files belonging to this database system will be owned by user "postgres".
pgdb      | This user must also own the server process.
pgdb      | 
pgdb      | The database cluster will be initialized with locale "en_US.utf8".
pgdb      | The default database encoding has accordingly been set to "UTF8".
pgdb      | The default text search configuration will be set to "english".
pgdb      | 
pgdb      | Data page checksums are disabled.
pgdb      | 
pgdb      | fixing permissions on existing directory /var/lib/postgresql/data ... ok
pgdb      | creating subdirectories ... ok
pgdb      | selecting dynamic shared memory implementation ... posix
pgdb      | selecting default max_connections ... 100
pgdb      | selecting default shared_buffers ... 128MB
pgdb      | selecting default time zone ... Etc/UTC
pgdb      | creating configuration files ... ok
pgdb      | running bootstrap script ... ok
pgdb      | performing post-bootstrap initialization ... ok
pgdb      | syncing data to disk ... initdb: warning: enabling "trust" authentication for local connections
pgdb      | You can change this by editing pg_hba.conf or using the option -A, or
pgdb      | --auth-local and --auth-host, the next time you run initdb.
pgdb      | ok
pgdb      | 
pgdb      | 
pgdb      | Success. You can now start the database server using:
pgdb      | 
pgdb      |     pg_ctl -D /var/lib/postgresql/data -l logfile start
pgdb      | 
pgdb      | waiting for server to start....2023-08-21 20:26:19.895 UTC [48] LOG:  starting PostgreSQL 12.16 (Debian 12.16-1.pgdg120+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
pgdb      | 2023-08-21 20:26:19.896 UTC [48] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
pgdb      | 2023-08-21 20:26:19.905 UTC [49] LOG:  database system was shut down at 2023-08-21 20:26:19 UTC
pgdb      | 2023-08-21 20:26:19.909 UTC [48] LOG:  database system is ready to accept connections
pgdb      |  done
pgdb      | server started
pgdb      | 
pgdb      | /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
pgdb      | 
pgdb      | 2023-08-21 20:26:20.024 UTC [48] LOG:  received fast shutdown request
pgdb      | waiting for server to shut down....2023-08-21 20:26:20.027 UTC [48] LOG:  aborting any active transactions
pgdb      | 2023-08-21 20:26:20.029 UTC [48] LOG:  background worker "logical replication launcher" (PID 55) exited with exit code 1
pgdb      | 2023-08-21 20:26:20.029 UTC [50] LOG:  shutting down
pgdb      | 2023-08-21 20:26:20.044 UTC [48] LOG:  database system is shut down
pgdb      |  done
pgdb      | server stopped
pgdb      | 
pgdb      | PostgreSQL init process complete; ready for start up.
pgdb      | 
pgdb      | 2023-08-21 20:26:20.139 UTC [1] LOG:  starting PostgreSQL 12.16 (Debian 12.16-1.pgdg120+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
pgdb      | 2023-08-21 20:26:20.139 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
pgdb      | 2023-08-21 20:26:20.139 UTC [1] LOG:  listening on IPv6 address "::", port 5432
pgdb      | 2023-08-21 20:26:20.142 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
pgdb      | 2023-08-21 20:26:20.150 UTC [68] LOG:  database system was shut down at 2023-08-21 20:26:20 UTC
pgdb      | 2023-08-21 20:26:20.153 UTC [1] LOG:  database system is ready to accept connections
platform  | Reset existing database
platform  | ----------------------------------------------------------
platform  | Database dropped and recreated successfully!
platform  | Create database migrations
platform  | ----------------------------------------------------------
platform  | No changes detected
platform  | Apply database migrations
platform  | ----------------------------------------------------------
platform  | Operations to perform:
platform  |   Synchronize unmigrated apps: debug_toolbar, django_extensions, django_jinja, formtools, messages, staticfiles
platform  |   Apply all migrations: admin, auth, commerce, contenttypes, engagement, product_management, security, sessions, talent
platform  | Synchronizing apps without migrations:
platform  |   Creating tables...
platform  |     Running deferred SQL...
platform  | Running migrations:
platform  |   Applying commerce.0001_initial... OK
platform  |   Applying security.0001_initial... OK
platform  |   Applying contenttypes.0001_initial... OK
platform  |   Applying admin.0001_initial... OK
platform  |   Applying admin.0002_logentry_remove_auto_add... OK
platform  |   Applying admin.0003_logentry_add_action_flag_choices... OK
platform  |   Applying contenttypes.0002_remove_content_type_name... OK
platform  |   Applying auth.0001_initial... OK
platform  |   Applying auth.0002_alter_permission_name_max_length... OK
platform  |   Applying auth.0003_alter_user_email_max_length... OK
platform  |   Applying auth.0004_alter_user_username_opts... OK
platform  |   Applying auth.0005_alter_user_last_login_null... OK
platform  |   Applying auth.0006_require_contenttypes_0002... OK
platform  |   Applying auth.0007_alter_validators_add_error_messages... OK
platform  |   Applying auth.0008_alter_user_username_max_length... OK
platform  |   Applying auth.0009_alter_user_last_name_max_length... OK
platform  |   Applying auth.0010_alter_group_name_max_length... OK
platform  |   Applying auth.0011_update_proxy_permissions... OK
platform  |   Applying auth.0012_alter_user_first_name_max_length... OK
platform  |   Applying product_management.0001_initial... OK
platform  |   Applying talent.0001_initial... OK
platform  |   Applying commerce.0002_initial... OK
platform  |   Applying engagement.0001_initial... OK
platform  |   Applying product_management.0002_initial... OK
platform  |   Applying product_management.0003_remove_product_owner... OK
platform  |   Applying product_management.0004_product_owner... OK
platform  |   Applying product_management.0005_delete_challengelisting... OK
platform  |   Applying security.0002_initial... OK
platform  |   Applying security.0003_user_password_reset_required_and_more... OK
platform  |   Applying security.0004_signinattempt_alter_user_managers... OK
platform  |   Applying security.0005_signinattempt_successful_signinattempt_user_and_more... OK
platform  |   Applying security.0006_alter_productperson_role... OK
platform  |   Applying security.0007_productroleassignment_delete_productperson... OK
platform  |   Applying security.0008_remove_signuprequest_email_and_more... OK
platform  |   Applying sessions.0001_initial... OK
platform  |   Applying talent.0002_remove_person_is_test_user_person_full_name_and_more... OK
platform  |   Applying talent.0003_person_github_link_person_website_link_and_more... OK
platform  |   Applying talent.0004_alter_person_user... OK
platform  | Load Sample Data
platform  | ----------------------------------------------------------
.....Complete!.................................................................Starting server
platform  | ----------------------------------------------------------
platform  | Watching for file changes with StatReloader
platform  | Performing system checks...
platform  | 
platform  | System check identified no issues (0 silenced).
platform  | August 21, 2023 - 20:27:10
platform  | Django version 4.2.2, using settings 'openunited.settings'
platform  | Starting development server at http://0.0.0.0:80/
platform  | Quit the server with CONTROL-C.
platform  |

**Review by adrianmcphee — APPROVED** [review]

**Comment by Saif-Uz-Zaman** [comment]
Hi @adrianmcphee the application runs on default http port 80, but I also exposed it on 8080. So after running the docker you have to hit on http://localhost:8080.

**Comment by adrianmcphee** [comment]
Thanks, there it is!

Screenshot 2023-08-21 at 22 44 04

**Pull request merged**